### ISSUE TITLE
ruby %alias on functions

### ISSUE THREAD

**Issue by theldoria** [swig/swig#138]
It is not possible to apply %alias to native C functions. Code generation silently failes (with swigwin-2.0.6).

Example:
mod.i

```
  %module mod

  %{
  #include "mod.h"
  %}

  %alias foo "bar";
  %include "mod.h"
```

mod.h

```
  #define bar foo
  extern int foo(int i);
```

mod.c

```
  #include "mod.h"
  int foo(int i)
  {
     return i + 1;
  }
```

**Comment by ojwb** [comment]
Still happens with current git master:

```
$ swig -ruby mod.i
Segmentation fault
```

gdb shows that `klass` is `NULL` here:

```
              Printv(klass->init, tab4, "rb_define_alias(", klass->vname, ", \"", alias.item, "\", \"", iname, "\");\n", NIL);
```

I don't know enough about Ruby to know where this needs to be emitted when there's no current class though.  @wsfulton Is there anyone actively maintaining SWIG/Ruby currently?

**Comment by goatshriek** [comment]
I've created a simple fix for this in Pull Request #1213 which adds support for the case where CPP is not in use. I also include three test cases for function aliases in the case of CPP methods (which was working before) as well as global functions and Ruby module functions (which both caused this issue).

**Comment by wsfulton** [comment]
Fixed by #1213.

@ojwb There is no active Ruby maintainer atm, but I'm comfortable enough with Ruby to review patches and merge them across.
